### ISSUE TITLE
Feature request: Change MANGLE_TARGET_NAMES to be the default

### ISSUE THREAD

**Issue by zbeekman** [kmorel/UseLATEX#6]
Motivation
========
While the short target names are convenient to type, the possibility of target name clashes might require patching upstream/vendor supplied `CMakeLists.txt` files. For example, consider the case where one wishes to distribute a document from a vendor or upstream source along with the main document in question, and the upstream/vendor source is considerate enough to use UseLATEX and CMake for compilation. Since CMake allows multiple [project()](http://www.cmake.org/cmake/help/v2.8.11/cmake.html#command:project) declarations (subprojects), the external document could be dumped into a sub-directory of the current project, say with a [subtree merge](https://help.github.com/articles/working-with-subtree-merge), and then [add_subdirectory()](http://www.cmake.org/cmake/help/v2.8.11/cmake.html#command:add_subdirectory) can be called to add the targets defined in the external document to the current project. (The `EXCLUDE_FROM_ALL` option can even be passed to make building the external reference or supplementary documentation optional/off by default.) With this setup, the auxiliary/reference/external document exists as it's own project and as part of the new project. Now, given that `MANGLE_TARGET_NAMES` is not the default, the [add_subdirectory()](http://www.cmake.org/cmake/help/v2.8.11/cmake.html#command:add_subdirectory) will cause a target name conflict. (The pdf target defined by the external document will clash with the pdf target of the current project.) The author (non-upstream/vendor) could enable `MANGLE_TARGET_NAMES` for his document(s) but when he distributes his document, packaged with the upstream/external document, users might expect that `make pdf` will build a pdf file from the main project, and be confused or surprised when they build the auxiliary/external project instead.

This could be beneficial in the context of subprojects too. For example, if one is writing a long PhD thesis or a book they could setup subprojects for each chapter that would then provide targets to build individual chapters instead of the entire document. However, in this context the author can easily choose to turn on name mangling, at the expense of a little more verbosity.

Pros
====
- No patching upstream CMakeLists.txt files or 
- surprising the end user when `make pdf` builds the 'wrong' document

Cons
====
- All users of the proposed new UseLATEX will be surprised when they type `make <document_type>` and no such target exists (it is now mangled by default: `basename_<document_type>`)
- This could also break some scripted builds etc.

Bottom Line
=========
I personally feel that this would be an improvement, but I also understand that the proposed change could be disruptive for some users.

**Comment by kmorel** [comment]
I've recently released UseLATEX.cmake 2.0, and this version changes the target behavior to be always mangled.
